lib/adal.js

```diff
@@ -788,7 +791,10 @@ AuthenticationContext.prototype.handleWindowCallback = function () {
         }
 
         window.location.hash = '';
-        window.location = this._getItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST);
+        if (this.config.navigateToLoginRequestUrl) {
```

Comment:
If the user sets navigateToLoginRequestUrl to false, then after the redirect, the token will be left in the hash part of the fragment which is not safe. We would just like some additional checks here:
1. Do not set window.location if we are already on the same page to prevent an app reload. 
2. In other cases, just set the hash part of the url to null.
Please refer to the snippet below for your reference:
  if (window.location.href.indexOf('#') != -1)
                    var urlWithoutHash = window.location.href.split("#")[0];
                else
                    var urlWithoutHash = window.location.href;
                if (this.config.navigateToLoginRequestUrl && urlWithoutHash !== this._getItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST))
                    window.location = this._getItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST);
                else
                    window.location.hash = '';